src/libopensc/pkcs15-prkey.c

```diff
@@ -234,6 +239,14 @@ int sc_pkcs15_decode_prkdf_entry(struct sc_pkcs15_card *p15card,
 	}
 	else if (asn1_prkey[1].flags & SC_ASN1_PRESENT) {
 		obj->type = SC_PKCS15_TYPE_PRKEY_EC;
+#ifdef ENABLE_OPENSSL
+		if (!info.field_length && ec_domain_len) {
+			const unsigned char *p = ec_domain;
+			const EC_GROUP *group = EC_GROUP_new_by_curve_name(OBJ_obj2nid(d2i_ASN1_OBJECT(NULL, &p, ec_domain_len)));
```

Comment:
please, check return values one after the other. This is short if it works, but debugging this line would be pain.

Reply:
Done.